MetaMask/metamask-extension
# Fixes for converting an ERC-20 token to an NFT

## Explanation

Fixes 2 issues when converting an ERC-20 token to an NFT:
- The contract address was not pre-populated in the modal
- The NFT was not removed from the tokens tab

This may have been a regression.  The NFT import modal already looked for a contract address + token removal indicator in `window.history`, but callers [1](https://github.com/MetaMask/metamask-extension/pull/19806/files#diff-2296a08979d8dc2053c60cc8e7c883d2054fc02592cf1e790f274c16e7cdd29fL42) and [2](https://github.com/MetaMask/metamask-extension/pull/19806/files#diff-4d87baa5c1671beca2baa2d8a456d56a581908e26d8acddc3a3fb9a4b413f1b5L318) stopped setting the `history` state.  Instead of `history`, I am now using redux state to pass data to the modal.  But I'm open to other suggestions for passing data to modals.

## Screenshots/Screencaps

### Before

During import:

https://github.com/MetaMask/metamask-extension/assets/3500406/89a93706-c80f-4e03-9b95-cb1745f1610d

Already imported (+ NFT remains on tokens tab):

https://github.com/MetaMask/metamask-extension/assets/3500406/7af96407-cc5d-44a1-b3cb-a7ba33daf263

### After

During import:

https://github.com/MetaMask/metamask-extension/assets/3500406/6ece1d55-0ee6-4789-9ecd-bb320036a493

Already imported (+ NFT removed from tokens tab):

https://github.com/MetaMask/metamask-extension/assets/3500406/e07c56f5-9abc-4707-b60e-6e8a5fb21ca6

## Manual Testing Steps

Try to import an NFT contract address as a token.  When prompted to convert to an NFT, the address should be pre-populated.

## Pre-merge author checklist


## Pre-merge reviewer checklist
